### PR TITLE
Eslint Configuration

### PR DESCRIPTION
**Pull request by asishap**
# Description

Added ESlint and prettier configuration

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      'anonymous': 'never',
+      'named': 'never'
+    }],
+    'radix': 'off',
+    'func-names': ['error', 'as-needed'],
+    'quotes': ['error', 'single', { 'allowTemplateLiterals': true }],
+    'comma-dangle': ['error', 'always-multiline'],
+    'semi': 'error',
+    'no-extra-semi': 'error',
+    'indent': ['error', 2, { "SwitchCase": 1 }],
```
Comment (ocgordon) [inline .eslintrc.js]: This is an opinionated configuration. Can we remove this and just rely on defaults here?

File: .eslintignore [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
```
Comment (ocgordon) [inline .eslintignore]: Please remove empty lines at the beginning of files

File: .eslintignore [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+node_modules
+mochawesome-report
+zkp/src/vkIds.json
+zkp/migrations
+zkp/code
+zkp-utils/node_modules
+offchain/migrations
+ui
```
Comment (ocgordon) [inline .eslintignore]: Please add a new line at the end of every file

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    node: true,
+  },
+  rules: {
+    'no-console': 'off',
+    'space-before-function-paren': ['error', {
+      'asyncArrow': 'always',
+      'anonymous': 'never',
+      'named': 'never'
+    }],
+    'radix': 'off',
```
Comment (ocgordon) [inline .eslintrc.js]: We need to work to remove these exceptions over time.

File: package.json [hunk 5]
```diff
@@ -9,6 +9,7 @@
     "fix": "npm run format && npm run lint -- --fix",
     "format": "prettier --write \"**/*.{json,css,scss,md,html}\"",
     "lint": "eslint .",
+    "ui:lint": "cd ui/ && npm ci && cd ../ && tslint -p ./ui/src/tsconfig.app.json",
```
Comment (ocgordon) [inline package.json]: Why does this line not match the format of the next one? It jumps from `ui:lint` to `lint:md`. And what is this line accomplishing? It looks like it switches to another directory, installs something, and switches back.

Comment (asishap) [reply package.json]: @ocgordon ,
We are using angular for UI, here typescript is the programming language. By default angular is integrated with typescript linting rules . In order to run the typescript linting,Angular depends on "codelyzer": "^4.0.1",  so we have to install dependencies before running lint command.

File: .eslintrc.js (hunk 4 above)
Comment (asishap) [reply .eslintrc.js]: Radix off rule is removed

File: package.json (hunk 5 above)
Comment (asishap) [reply package.json]: Added spaces where ever required

**Pull request merged**